pkg/status/status.go

```diff
@@ -175,52 +197,60 @@ func (m *Prober) IsProbeActive(ing *v1alpha1.Ingress) (ProbeState, bool) {
 	return state, ok
 }
 
-// IsReady checks if the provided Ingress is ready, i.e. the Envoy pods serving the Ingress
-// have all been updated. This function is designed to be used by the Ingress controller, i.e. it
-// will be called in the order of reconciliation. This means that if IsReady is called on an Ingress,
-// this Ingress is the latest known version and therefore anything related to older versions can be ignored.
-// Also, it means that IsReady is not called concurrently.
-func (m *Prober) IsReady(ctx context.Context, ing *v1alpha1.Ingress) (bool, error) {
-	ingressKey := types.NamespacedName{Namespace: ing.Namespace, Name: ing.Name}
-	logger := logging.FromContext(ctx)
-
-	bytes, err := ingress.ComputeHash(ing)
-	if err != nil {
-		return false, fmt.Errorf("failed to compute the hash of the Ingress: %w", err)
-	}
-	hash := fmt.Sprintf("%x", bytes)
-
-	if ready, ok := func() (bool, bool) {
+func (m *Prober) DoProbes(ctx context.Context, backends Backends) (ProbeState, error) {
+	if state, ok := func() (ProbeState, bool) {
 		m.mu.Lock()
 		defer m.mu.Unlock()
-		if state, ok := m.ingressStates[ingressKey]; ok {
-			if state.version == hash {
+		if state, ok := m.ingressStates[backends.Key]; ok {
```

Comment:
maybe name `state` `ingState` to make it more clear compared to e.g. `probeState`?

Reply:
done